app/wallet/app.js

```diff
@@ -5,3 +5,20 @@ export const onAppReloadRequested = cb => ipcRenderer.on("app-reload-requested",
 export const log = (level, ...args) => {
   ipcRenderer.send("main-log", ...[level, ...args]);
 };
+
+// Higher Order Function that wraps the promise-generating function f with log
+// calls. Calling f() **must** return a promise.
+export const withLog = (msg, f) => (...args) => {
+  log("info", "Wallet requesting %s", msg);
+  return new Promise((resolve, reject) => {
+    f(...args)
+      .then((...res) => {
+        log("debug", "Wallet received response %s", ...res);
```

Comment:
Can we add the msg here too, so we can easily see what response this corresponds.

Reply:
Also on most of my responses it didn't resolve the `%s` and just showed it at `Wallet received response %s`

When it did resolve it showed [Object object].

I'm guessing this is due to most of these loader requests not having any `resolve(response)`.